wolfcrypt/src/evp.c

```diff
@@ -4321,47 +4322,53 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD *md)
                 break;
 #if !defined(_WIN32) && !defined(HAVE_FIPS)
             case EVP_CTRL_GCM_IV_GEN:
+                if ((ctx->flags & WOLFSSL_EVP_CIPH_FLAG_AEAD_CIPHER) == 0)
+                    break;
                 if (ctx->cipher.aes.keylen == 0 || ctx->ivSz == 0) {
-                    ret = WOLFSSL_FAILURE;
                     WOLFSSL_MSG("Key or IV not set");
                     break;
                 }
                 if ((ret = wc_AesGcmSetExtIV(&ctx->cipher.aes, ctx->iv,
```

Comment:
We can't return this value directly. OpenSSL returns 0 or 1 while we return negative on error. A user negating the return for error checking assumes anything non-zero means success. If you don't store the error returned by `wc_AesGcmSetExtIV` then `break`'ing out will be a valid way to handle this error.

Reply:
fixed in 785a8f8 -- thanks again for catching that!